pkgs/development/python-modules/gradio/default.nix

```diff
@@ -0,0 +1,204 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, writeText
+
+# pyproject
+, hatchling
+, hatch-requirements-txt
+, hatch-fancy-pypi-readme
+
+# runtime
+, aiofiles
+, aiohttp
+, altair
+, fastapi
+, ffmpy
+, markdown-it-py
+, mdit-py-plugins
+, markupsafe
+, matplotlib
+, numpy
+, orjson
+, pandas
+, pillow
+, pycryptodome
+, python-multipart
+, pydub
+, pyyaml
+, requests
+, uvicorn
+, jinja2
+, fsspec
+, httpx
+, pydantic
+, websockets
+, typing-extensions
+
+# check
+, pytestCheckHook
+, pytest-asyncio
+, mlflow
+, huggingface-hub
+, transformers
+, wandb
+, respx
+, scikit-image
+, shap
+, ipython
+, ffmpeg
+, vega_datasets
+, boto3
+}:
+
+let
+  # breaks way too frquently
+  shapWorks = (builtins.tryEval shap.outPath).success;
+in
+
+buildPythonPackage rec {
+  pname = "gradio";
+  version = "3.20.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  # We use the Pypi release, as it provides prebuild webui assets,
+  # and its releases are also more frequent than github tags
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-oG97GwehyBWjWXzDqyfj+x2mAfM6OQhYKdA3j0Rv8Vs=";
+  };
+
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace \
+        "mdit-py-plugins<=0.3.3" \
+        "mdit-py-plugins"
```

Comment:
```suggestion
      --replace "mdit-py-plugins<=0.3.3" "mdit-py-plugins"
```

Reply:
Could we keep the formatting as is ? The multi-line makes it easy to verify there are no typos.

Reply:
isn't that what `pythonRelaxDeps = [ ];` is for?

Reply:
It works now? That is news to me

Reply:
It worked for me months ago, but you have to be careful with the capitalization and `-` and so on in the strings and can't just use the pname of the dep.

Reply:
I didn't know this hook was working with pyproject.
I tested this overnight however and it does work. I used exactly `mdit-py-plugins` as the dependency to relax.

Reply:
Then I suggest we switch to this since it's a "cleaner" way :D

Reply:
That works for me . @pbsds ill let you make the change when you come back

Reply:
Doesn't work on my end